Library/Homebrew/tap.rb

```diff
@@ -820,7 +820,7 @@ def formula_reverse_renames
 
   sig { returns(T::Hash[String, T::Array[String]]) }
   def self.reverse_tap_migrations_renames
-    Tap.each_with_object({}) do |tap, hash|
+    cache[:reverse_tap_migrations_renames] ||= Tap.each_with_object({}) do |tap, hash|
```

Comment:
@apainintheneck, I remember specifically not caching this because it will never be cleared.

Basically, every instance's `#clear_cache` would need to clear the global `cache[:reverse_tap_migrations_renames]`, because it depends on all taps.

Reply:
My understanding is that `Tap.reverse_tap_migrations_renames` loads each tap with `Tap.each` and then calls `Tap#tap_migrations` on each tap to build this list. Do we expect either `Tap.each` or `Tap#tap_migrations` to change over the life of the program?

Reply:
I guess you're worried about the case where a tap gets installed with `Tap#ensure_installed?` after `Tap.reverse_tap_migrations_renames` has been built, right? Maybe it'd just be a good idea to clear the class level cache when we do that since `Tap#ensure_installed!` should only very rarely need to install a new tap.

Reply:
Or we could just go back to how things are. Either way works for me.

Reply:
Could just have the cache inside the `Tap.each_with_object` instead and cache by `cache[:reverse_tap_migrations_renames][tap.name]`

Reply:
Now that I look at it a bit more the problem might be `Tap.each` in this case.

![Screen Shot 2024-03-02 at 10 51 57 AM](https://github.com/Homebrew/brew/assets/42982186/8cd46374-f540-4975-b9b3-c2efe2842ff2)

This is from running `brew prof -- deps --casks --eval-all`.

Reply:
A special cache of installed tap instances that gets reset on `Tap#install` and `Tap#uninstall` would make sense to me.

Reply:
That sounds doable. I'm not sure how that could be implemented in a simple, straightforward way though. This seemed like a no-brainer but now I know it wasn't.

Reply:
> A special cache of installed tap instances that gets reset on `Tap#install` and `Tap#uninstall` would make sense to me.

@Bo98 @apainintheneck @reitermarkus I think we should/could just nuke _all_ tap caches on those methods given how infrequently they are calls.